src/utils/string_to_enum.C

```diff
@@ -97,566 +100,480 @@ void build_reverse_map (MapIter it, MapIter end, MapType & reverse)
+   {"THIRD"        , THIRD},
+   {"FOURTH"       , FOURTH},
+   {"FIFTH"        , FIFTH},
+   {"SIXTH"        , SIXTH},
+   {"SEVENTH"      , SEVENTH},
+   {"EIGHTH"       , EIGHTH},
+   {"NINTH"        , NINTH},
+   {"TENTH"        , TENTH},
+
+   {"ELEVENTH"     , ELEVENTH},
+   {"TWELFTH"      , TWELFTH},
+   {"THIRTEENTH"   , THIRTEENTH},
+   {"FOURTEENTH"   , FOURTEENTH},
+   {"FIFTEENTH"    , FIFTEENTH},
+   {"SIXTEENTH"    , SIXTEENTH},
+   {"SEVENTEENTH"  , SEVENTEENTH},
+   {"EIGHTTEENTH"  , EIGHTTEENTH},
+   {"NINETEENTH"   , NINETEENTH},
+   {"TWENTIETH"    , TWENTIETH},
+
+   {"TWENTYFIRST"  , TWENTYFIRST},
+   {"TWENTYSECOND" , TWENTYSECOND},
+   {"TWENTYTHIRD"  , TWENTYTHIRD},
+   {"TWENTYFOURTH" , TWENTYFOURTH},
+   {"TWENTYFIFTH"  , TWENTYFIFTH},
+   {"TWENTYSIXTH"  , TWENTYSIXTH},
+   {"TWENTYSEVENTH", TWENTYSEVENTH},
+   {"TWENTYEIGHTH" , TWENTYEIGHTH},
+   {"TWENTYNINTH"  , TWENTYNINTH},
+   {"THIRTIETH"    , THIRTIETH},
+
+   {"THIRTYFIRST"  , THIRTYFIRST},
+   {"THIRTYSECOND" , THIRTYSECOND},
+   {"THIRTYTHIRD"  , THIRTYTHIRD},
+   {"THIRTYFOURTH" , THIRTYFOURTH},
+   {"THIRTYFIFTH"  , THIRTYFIFTH},
+   {"THIRTYSIXTH"  , THIRTYSIXTH},
+   {"THIRTYSEVENTH", THIRTYSEVENTH},
+   {"THIRTYEIGHTH" , THIRTYEIGHTH},
+   {"THIRTYNINTH"  , THIRTYNINTH},
+   {"FORTIETH"     , FORTIETH},
+
+   {"FORTYFIRST"   , FORTYFIRST},
+   {"FORTYSECOND"  , FORTYSECOND},
+   {"FORTYTHIRD"   , FORTYTHIRD}
+  };
+
+std::map<Order, std::string> enum_to_order =
+  build_reverse_map(order_to_enum);
+
+
+std::map<std::string, FEFamily> fefamily_to_enum {
+   {"LAGRANGE"          , LAGRANGE},
+   {"LAGRANGE_VEC"      , LAGRANGE_VEC},
+   {"L2_LAGRANGE"       , L2_LAGRANGE},
+   {"L2_LAGRANGE_VEC"   , L2_LAGRANGE_VEC},
+   {"HIERARCHIC"        , HIERARCHIC},
+   {"L2_HIERARCHIC"     , L2_HIERARCHIC},
+   {"SIDE_HIERARCHIC"   , SIDE_HIERARCHIC},
+   {"MONOMIAL"          , MONOMIAL},
+   {"MONOMIAL_VEC"      , MONOMIAL_VEC},
+   {"SCALAR"            , SCALAR},
+   {"XYZ"               , XYZ},
+   {"BERNSTEIN"         , BERNSTEIN},
+   {"RATIONAL_BERNSTEIN", RATIONAL_BERNSTEIN},
+   {"SZABAB"            , SZABAB},
+   {"INFINITE_MAP"      , INFINITE_MAP},
+   {"JACOBI_20_00"      , JACOBI_20_00},
+   {"JACOBI_30_00"      , JACOBI_30_00},
+   {"LEGENDRE"          , LEGENDRE},
+   {"CLOUGH"            , CLOUGH},
+   {"HERMITE"           , HERMITE},
+   {"SUBDIVISION"       , SUBDIVISION},
+   {"NEDELEC_ONE"       , NEDELEC_ONE},
+   {"RAVIART_THOMAS"    , RAVIART_THOMAS},
+   {"L2_RAVIART_THOMAS" , L2_RAVIART_THOMAS}
+  };
+
+std::map<FEFamily, std::string> enum_to_fefamily =
+  build_reverse_map(fefamily_to_enum);
+
+
+std::map<std::string, InfMapType> inf_map_type_to_enum {
+   {"CARTESIAN"  , CARTESIAN},
+   {"SPHERICAL"  , SPHERICAL},
+   {"ELLIPSOIDAL", ELLIPSOIDAL}
+  };
+
+std::map<InfMapType, std::string> enum_to_inf_map_type =
+  build_reverse_map(inf_map_type_to_enum);
+
+
+std::map<std::string, QuadratureType> quadrature_type_to_enum {
```

Comment:
It looks like we are missing QMONOMIAL

Reply:
So we are!  In my defense, we were missing it before too.  I started out by hand-editing these things but quickly decided I would introduce fewer errors if I did it all via regexps; didn't consider that I'd *remove* fewer errors that way too.

I'll fix it and the indentation now.